coffee-model/coffee-model-base/src/main/java/hu/icellmobilsoft/coffee/model/base/audit/AuditProvider.java

```diff
@@ -19,89 +19,115 @@
  */
 package hu.icellmobilsoft.coffee.model.base.audit;
 
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.List;
 import java.util.Set;
 
+import hu.icellmobilsoft.coffee.model.base.AbstractProvider;
+import hu.icellmobilsoft.coffee.model.base.annotation.CreatedBy;
+import hu.icellmobilsoft.coffee.model.base.annotation.CurrentUser;
+import hu.icellmobilsoft.coffee.model.base.annotation.ModifiedBy;
+import hu.icellmobilsoft.coffee.model.base.exception.ProviderException;
 import jakarta.enterprise.context.Dependent;
 import jakarta.enterprise.inject.spi.Bean;
 import jakarta.enterprise.inject.spi.BeanManager;
 import jakarta.inject.Inject;
-
-import org.apache.deltaspike.core.util.metadata.AnnotationInstanceProvider;
-import org.apache.deltaspike.data.api.audit.CurrentUser;
-import org.apache.deltaspike.data.impl.audit.AuditPropertyException;
-import org.apache.deltaspike.data.impl.audit.PrePersistAuditListener;
-import org.apache.deltaspike.data.impl.audit.PreUpdateAuditListener;
-import org.apache.deltaspike.data.impl.property.Property;
-import org.apache.deltaspike.data.impl.property.query.AnnotatedPropertyCriteria;
-import org.apache.deltaspike.data.impl.property.query.PropertyQueries;
-import org.apache.deltaspike.data.impl.property.query.PropertyQuery;
-
-import hu.icellmobilsoft.coffee.model.base.annotation.CreatedBy;
-import hu.icellmobilsoft.coffee.model.base.annotation.ModifiedBy;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
 
 /**
- * Perist entity @CreatedBy property before persist with the value provided by @CurrentUser
+ * Persist entity @CreatedBy property before persist with the value provided by @CurrentUser
  *
  * Update entity @ModifiedBy property before update with the value provided by @CurrentUser
  *
  * @author czenczl
+ * @author zsolt.vasi
  * @since 1.0.0
  */
 @Dependent
-public class AuditProvider implements PrePersistAuditListener, PreUpdateAuditListener {
+public class AuditProvider extends AbstractProvider {
 
     @Inject
     private BeanManager manager;
 
-    /** {@inheritDoc} */
-    @Override
+    /**
+     * Persist entity @CreatedBy (@ModifiedBy) property before persist with the value provided by @CurrentUser
+     *
+     * @param entity
+     *            Object entity to persist
+     */
+    @PrePersist
     public void prePersist(Object entity) {
-        PropertyQuery<Object> createdByQuery = PropertyQueries.<Object> createQuery(entity.getClass())
-                .addCriteria(new AnnotatedPropertyCriteria(CreatedBy.class));
-        for (Property<Object> property : createdByQuery.getWritableResultList()) {
-            Object value = resolvePrincipal(entity, property);
-            setProperty(entity, property, value);
+        List<Field> allFields = getAllFields(entity.getClass());
+        for (Field field : allFields) {
```

Comment:
gyorsitas miatt nem kellene valami cache-be rakni ezeket a field es metodusokat? Nem valtozik menet kozben es gyakori lehet nagyon az ilyen scan

Reply:
a megbeszelteknek megfeleloen bevezettem ra egy cache-t